sympy/printing/tests/test_str.py

```diff
@@ -666,14 +666,14 @@ def test_settings():
 def test_RandomDomain():
     from sympy.stats import Normal, Die, Exponential, pspace, where
     X = Normal('x1', 0, 1)
-    assert str(where(X > 0)) == "Domain: And(0 < x1, x1 < oo)"
+    assert str(where(X > 0)) == "Domain: (0 < x1 & x1 < oo)"
 
     D = Die('d1', 6)
-    assert str(where(D > 4)) == "Domain: Or(Eq(d1, 5), Eq(d1, 6))"
+    assert str(where(D > 4)) == "Domain: (Eq(d1, 5) | Eq(d1, 6))"
 
     A = Exponential('a', 1)
     B = Exponential('b', 1)
-    assert str(pspace(Tuple(A, B)).domain) == "Domain: And(0 <= a, 0 <= b, a < oo, b < oo)"
+    assert str(pspace(Tuple(A, B)).domain) == "Domain: (0 <= a & 0 <= b & a < oo & b < oo)"
```

Comment:
> & is Bitwise AND in python and has higher precedence than relational operators.

Yes, that's the problem. You need parentheses.

> Shall I replace it with and which is python's Boolean AND and has lower precedence than relational operators?

No, `and` is not supposed to return `And` objects, it should be evaluated and immediately return a result. `&` is supposed to return unevaluated `And` expressions in case the truthness cannot be immediately determined.